src/utils/messages.ts

```diff
@@ -143,3 +148,42 @@ export const retrievedMessageToPublic = (
   sender_service_id: retrievedMessage.senderServiceId
 });
 /* eslint-enable @typescript-eslint/naming-convention */
+
+/**
+ * This function enrich a CreatedMessageWithoutContent with
+ * service's details and message's subject.
+ *
+ * @param messageModel
+ * @param serviceModel
+ * @param blobService
+ * @returns
+ */
+export const enrichMessageData = (
+  messageModel: MessageModel,
+  serviceModel: ServiceModel,
+  blobService: BlobService
+) => (
+  message: CreatedMessageWithoutContent
+): Promise<E.Either<Error, EnrichedMessage>> =>
+  pipe(
+    TE.Do,
+    TE.bind("service", () =>
+      serviceModel.findLastVersionByModelId([message.sender_service_id])
+    ),
+    TE.mapLeft(E.toError),
+    TE.bind("messageContent", () =>
+      messageModel.getContentFromBlob(blobService, message.id)
+    ),
+    TE.map(x => {
+      const content = O.getOrElse(() => ({} as MessageContent))(
+        x.messageContent
+      );
+      const service = O.getOrElse(() => ({} as Service))(x.service);
+      return {
+        ...message,
+        message_title: content.subject,
+        organization_name: service.organizationName,
+        service_name: service.serviceName
+      };
+    })
+  )();
```

Comment:
We could get rid of `x` variable

```suggestion
  pipe(
    TE.Do,
    TE.bind("maybeService", () =>
      serviceModel.findLastVersionByModelId([message.sender_service_id])
    ),
    TE.mapLeft(E.toError),
    TE.bind("maybeMessageContent", () =>
      messageModel.getContentFromBlob(blobService, message.id)
    ),
    TE.map({maybeService, maybeMessageContent} => {
      const content = O.getOrElse(() => ({} as MessageContent))(
        maybeMessageContent
      );
      const service = O.getOrElse(() => ({} as Service))(maybeService);
      return {
        ...message,
        message_title: content.subject,
        organization_name: service.organizationName,
        service_name: service.serviceName
      };
    })
  )();
```

Reply:
Applied, but on fn-app repo.